src/backend/distributed/commands/alter_table.c

```diff
@@ -368,7 +368,7 @@ UndistributeTable(TableConversionParameters *params)
 		EnsureTableNotReferencing(params->relationId, UNDISTRIBUTE_TABLE);
 		EnsureTableNotReferenced(params->relationId, UNDISTRIBUTE_TABLE);
 	}
-	EnsureTableNotForeign(params->relationId);
```

Comment:
I think we have some issues with user mapping after undistribute:
```SQL

CREATE FOREIGN TABLE foreign_table (
	a int 
)       SERVER foreign_server
        OPTIONS ( table_name 'other_table');


CREATE USER MAPPING FOR CURRENT_USER
        SERVER foreign_server
        OPTIONS (user 'onderkalaci');
        SELECT citus_add_local_table_to_metadata('foreign_table');

select count(*) from foreign_table;
NOTICE:  executing the command locally: SELECT count(*) AS count FROM public.foreign_table_102046 foreign_table
 count 
-------
 10001
 select undistribute_table('foreign_table');           
NOTICE:  creating a new table for public.foreign_table
NOTICE:  extension "postgres_fdw" already exists, skipping
NOTICE:  server "foreign_server_2537643067" already exists, skipping
NOTICE:  dropping the old public.foreign_table
NOTICE:  renaming the new table to public.foreign_table
 undistribute_table 
--------------------
 
(1 row)

Time: 120.020 ms
 select count(*) from foreign_table;
ERROR:  user mapping not found for "onderkalaci"
Time: 0.794 ms

```

Reply:
it's probably because of the server, not the mapping.
Still investigating

Reply:
fixed, thanks.
Pushed with tests